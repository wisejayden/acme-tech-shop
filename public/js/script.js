(function() {

		var container = document.getElementById( 'container' ),
			trigger = container.querySelector( 'button.catalog' );


		function toggleContent() {
			if( classie.has( container, 'container--open' ) ) {
				classie.remove( container, 'container--open' );
				classie.remove( trigger, 'trigger--active' );
				window.addEventListener( 'scroll', noscroll );
			}
			else {
				classie.add( container, 'container--open' );
				classie.add( trigger, 'trigger--active' );
				window.removeEventListener( 'scroll', noscroll );
			}
		}

		function noscroll() {
			window.scrollTo( 0, 0 );
		}

		// reset scrolling position
		document.body.scrollTop = document.documentElement.scrollTop = 0;

		// disable scrolling
		window.addEventListener( 'scroll', noscroll );

		trigger.addEventListener( 'click', toggleContent {
			console.log("HELLo")
		} );

	})();
