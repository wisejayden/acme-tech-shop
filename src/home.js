import React from 'react';

export class Home extends React.Component {
    constructor(props) {
        super(props);
        this.state = {};

    }
    render() {
        return (
            <div className="splash">
                <img src="/images/desk-tech.jpg" className="splash-image" alt="Our Tech Splash" />
            </div>
        )
    }
}
