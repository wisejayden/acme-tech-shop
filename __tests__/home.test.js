import React from 'react';
import { shallow, mount, render } from 'enzyme';
import {Home} from '../src/home.js';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';


test('Home component should render as expected', () => {
    const component = shallow(<Home />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
})
