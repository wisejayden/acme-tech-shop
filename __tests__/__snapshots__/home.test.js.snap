// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should render as expected 1`] = `
<div>
  <div
    className="splash"
  >
    <img
      alt="Our Tech Splash"
      className="splash-image"
      src="/images/tech1.jpg"
    />
    <Link
      replace={false}
      to="/catalog"
    >
      <button
        className="splash-button"
        type="button"
      >
        SHOP NOW
      </button>
    </Link>
  </div>
  <div
    className="home-content"
  >
    <div
      className="home-content-box"
    >
      <h2>
        Lorem
      </h2>
      <p>
        An velit fabulas reprehenderit. In incididunt hic admodum. Singulis in export senserit est tempor te constias ad cillum.Labore non ullamco, ex an arbitrantur. Ea cillum praetermissum. Officia legam commodo consequat o mandaremus ad aute do quo o quid quis legam aut cupidatat veniam ingeniis cernantur. Fabulas o fabulas, sed nulla fore non mandaremus, ipsum incurreret ea familiaritatem, quae constias ubi duis malis, ea duis appellat ne non legam legam ut proident, iudicem illum excepteur cupidatat aut minim litteris se occaecat.
      </p>
    </div>
  </div>
</div>
`;
